glom/core.py

```diff
@@ -1323,13 +1323,35 @@ def _t_child(parent, operation, arg):
     return t
 
 
+def _s_first_magic(scope, key, _t):
+    """
+    enable S.a to do S['a'] or S['a'].val as a special
+    case for accessing user defined string variables
+    """
+    pae = None
+    try:
+        cur = scope[key]
+    except (KeyError, IndexError, TypeError) as e:
```

Comment:
can TypeError happen here? Like a `scope` might be passed in which doesn't support getitem/indexing?

Reply:
just copied from the main loop didn't think -- I don't think we need to worry about `scope` not being the right type